### DISCUSSION TITLE
libcurl: Proxying DNS requests

### DISCUSSION THREAD

**Discussion by t1m013y** [curl/curl#15866] [Q&A] [answered]
Will DNS requests made by libcurl be done through the proxy if the proxy is enabled?

**Comment by dfandrich** [comment]
curl supports many kinds of proxies. What kind are you interested in?

**Reply by t1m013y** [reply]
Socks4, Socks5

**Comment by dfandrich** [comment, marked as answer]
With SOCKS proxies, the `socks5h:` and `socks4a:` schemes will send the hostname to the proxy to do the resolving while the other SOCKS types will resolve hostnames locally.

**Reply by t1m013y** [reply]
Does "send the hostname to the proxy" mean DNS requests will be performed through proxy?

**Comment by dfandrich** [comment]
That's what it means.
